beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestAddonChangeAlignment.java

```diff
@@ -0,0 +1,192 @@
+
+        //ADDON CTD is 2023-08-05 as expected
+        Subscription addonSub = subscriptionApi.getSubscriptionForEntitlementId(addonEntId, false, callContext);
+        Assert.assertEquals(addonSub.getChargedThroughDate(), new LocalDate(2023, 8, 05));
+
+        clock.setDay(new LocalDate(2023, 8, 15));
+
+        //change addon with start of bundle alignment on 2023-08-15 - change will happen immediately as subscription is in TRIAL phase
+        busHandler.pushExpectedEvents(NextEvent.CHANGE, NextEvent.INVOICE);
+        final PlanPhaseSpecifier newAddOnSpec = new PlanPhaseSpecifier("BasicAOStartOfBundle-monthly");
+        addonSub.changePlan(new DefaultEntitlementSpecifier(newAddOnSpec), Collections.emptyList(), callContext);
+        assertListenerStatus();
+
+        //Base CTD is still 2023-08-01 as expected
+        baseSub = subscriptionApi.getSubscriptionForEntitlementId(baseEntId, false, callContext);
+        Assert.assertEquals(baseSub.getChargedThroughDate(), new LocalDate(2023, 8, 1));
+
+        //ADDON CTD is 2023-08-15, though I think technically this should be 2023-08-01 (aligned with base CTD), I think 2023-08-15 is acceptable
+        addonSub = subscriptionApi.getSubscriptionForEntitlementId(addonEntId, false, callContext);
+        Assert.assertEquals(addonSub.getChargedThroughDate(), new LocalDate(2023, 8, 15));
+
+        //move clock to 2023-08-31 - Phase change for BASE, ADDON takes effect
+        busHandler.pushExpectedEvents(NextEvent.PHASE, NextEvent.PHASE, NextEvent.INVOICE, NextEvent.INVOICE_PAYMENT, NextEvent.PAYMENT, NextEvent.NULL_INVOICE);
+        clock.setDay(new LocalDate(2023, 8, 31));
+        assertListenerStatus();
+
+        // BASE CTD set to 2023-09-01
+        baseSub = subscriptionApi.getSubscriptionForEntitlementId(baseEntId, false, callContext);
+        Assert.assertEquals(baseSub.getChargedThroughDate(), new LocalDate(2023, 9, 1));
+
+        // ADDON CTD is 2023-09-01
+        addonSub = subscriptionApi.getSubscriptionForEntitlementId(addonEntId, false, callContext);
+        Assert.assertEquals(addonSub.getChargedThroughDate(), new LocalDate(2023, 9, 1));
+
+        //move clock to 2023-09-30 - invoice for base and addon
+        busHandler.pushExpectedEvents(NextEvent.INVOICE, NextEvent.INVOICE_PAYMENT, NextEvent.PAYMENT);
+        clock.setDay(new LocalDate(2023, 9, 30));
+        assertListenerStatus();
+
+        // BASE CTD set to 2023-10-01
+        baseSub = subscriptionApi.getSubscriptionForEntitlementId(baseEntId, false, callContext);
+        Assert.assertEquals(baseSub.getChargedThroughDate(), new LocalDate(2023, 10, 1));
+
+        // ADDON CTD is 2023-10-01
+        addonSub = subscriptionApi.getSubscriptionForEntitlementId(addonEntId, false, callContext);
+        Assert.assertEquals(addonSub.getChargedThroughDate(), new LocalDate(2023, 10, 1));
+
+    }
+
+    @Test(groups = "slow")
+    public void testChangeAlignmentStartOfBundleImmediateChange() throws Exception {
```

Comment:
Is it true to say that this test will produce exactly same result as IMMEDIATE policy and EOT currently default to immediate?

Reply:
> Is it true to say that this test will produce exactly same result as IMMEDIATE policy and EOT currently default to immediate?

Yes, that is correct, changing the plan with `IMMEDIATE` policy will behave in the same way as changing the plan with default `EOT` policy (since `EOT` defaults to `IMMEDIATE` in `TRIAL` phase)